src/platform/utilities/feature-toggles/useFeatureToggle.js

```diff
@@ -1,27 +1,110 @@
-import { useSelector } from 'react-redux';
+import { useEffect, useMemo } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { setData } from '../../forms-system/src/js/actions';
 import TOGGLE_NAMES from './featureFlagNames';
 
 export const toggleValuesSelector = state => state.featureToggles || {};
 
 export const loadingTogglesSelector = state => state?.featureToggles?.loading;
 
-export const useFeatureToggle = () => {
-  const useToggleValue = toggleName => {
-    if (!Object.values(TOGGLE_NAMES).includes(toggleName)) {
-      throw new Error(
-        `Invalid toggle name "${toggleName}". Did you add the toggle name to featureFlagNames.js?`,
-      );
-    }
-    return useSelector(state => toggleValuesSelector(state)?.[toggleName]);
-  };
+/**
+ * Configuration type for a single feature toggle
+ * @typedef { Object } ToggleConfig
+ * @property { string } toggleName - CamelCased name of the feature toggle example: 'profileUseExperimental'
+ * @property { string } [formKey] - Optional custom key for form / session storage
+ */
 
-  const useToggleLoadingValue = () => {
-    return useSelector(state => loadingTogglesSelector(state));
-  };
+/**
+ * Get feature toggle loading state
+ * @returns {Boolean} - feature flag loading state
+ */
+export const useToggleLoadingValue = () => {
+  return useSelector(state => loadingTogglesSelector(state));
+};
+
+/**
+ * Hook to sync feature toggles with form data and session storage
+ * @param {(string|ToggleConfig)[]} toggles - Array of toggle name strings or ToggleConfig objects
+ * @param {Function} [setStorageItem] - Storage setter function (defaults to sessionStorage.setItem)
+ */
+export const useFormFeatureToggleSync = (
+  toggles,
+  setStorageItem = window.sessionStorage.setItem.bind(window.sessionStorage),
+) => {
+  const dispatch = useDispatch();
+  const formData = useSelector(state => state?.form?.data ?? {});
+  const isLoadingFeatureFlags = useToggleLoadingValue();
+  const featureToggles = useSelector(toggleValuesSelector);
+
+  // Normalize toggle configurations
+  const normalizedToggles = useMemo(
+    () =>
+      toggles.map(toggle => {
+        if (typeof toggle === 'string') {
+          return { toggleName: toggle, formKey: toggle };
+        }
+        return { ...toggle, formKey: toggle.formKey ?? toggle.toggleName };
+      }),
+    [toggles],
+  );
```

Comment:
This memoized function helps keep things consistent from the string vs ToggleConfig usage and also have a perf boost over using just a derived value with useMemo

Reply:
Awesome!